Fix browser console warnings for audio source elements

Harsheet-11 opened the pull request:
Fixes: #36371

### Background

I started working on this issue but did not realize someone else had claimed it yesterday.  
I have completed the implementation and testing. If @Shreyanshu005 has also made progress, I'm happy to coordinate, but I wanted to submit my working solution for review.

### Problem

Empty `<source>` elements in `index.html` caused browser console warnings on every page load:
- `Media resource load failed`
- `<source> element has no "src" attribute`

These warnings appeared in Firefox, as mentioned in the discussion [#issues > Zulip fails to reload in background tab (Firefox) #36094 @ 💬](https://chat.zulip.org/#narrow/channel/9-issues/topic/Zulip.20fails.20to.20reload.20in.20background.20tab.20.28Firefox.29.20.2336094/near/2283439)


### Solution

- Removed empty `<source>` tags from HTML.
- Create `<source>` elements dynamically in `update_notification_sound_source()`, so it only creates sources when we have a valid `src` to set.

This ensures sources are never created without a `src` attribute, eliminating the warnings.

### Changes

**`templates/zerver/app/index.html`:**
- Removed empty `<source>` elements from audio tags.

**`web/src/audible_notifications.ts`:**
- Modified `update_notification_sound_source()` to:
  - Clear existing sources with `$container_elem.empty()`.
  - Create new `<source>` elements with `src` already set.
  - Only create sources when `notification_sound !== "none"`.
  - Append sources to audio element before calling `load()`.

### Testing

- [x] Linter passes.
- [x] All JavaScript tests pass (`./tools/test-js-with-node`).
- [x] No console warnings on page load (tested in Firefox and Chrome).
- [x] Notification sounds play correctly when clicking the "Play" button.
- [x] Changing notification sound in settings works correctly.
- [x] Setting sound to "None" properly empties the audio element.

### Screenshots :
                                                             
 ### BEFORE
<img width="2047" height="1278" alt="Before fix warning messages" src="https://github.com/user-attachments/assets/1e0a8924-6df0-42d0-8766-8f696f9f9634" />


 ### AFTER
<img width="2047" height="1278" alt="after fix warning messages in console" src="https://github.com/user-attachments/assets/d4b442bc-0b56-4267-88ad-534b40cd55cf" />


<details>
<summary>Self-review checklist</summary>

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
You'll need to address CodeQL comments above (are they correct? fix if so, or explain why not).

Commit:
notifications: Dynamically create audio sources to prevent browser warnings

Previously, we would have empty  tags without src attributes in index.html, which generated warnings in Firefox and Chrome.
Audio elements are created dynamically only when a valid notification sound has been selected, and the src is set immediately.
The notification_sound value is sanitized, and we verify that the audio element exists to make the code more robust and reliable.

Fixes #36371.

alya commented:
@Harsheet-11 I recommend that you review our recently documented [AI use policy](https://zulip.readthedocs.io/en/latest/contributing/contributing.html#ai-use-policy-and-guidelines).
